### PR TITLE
add new coredns template for v1.8.3

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/31567

### CODE REVIEW

**Review by StrongMonkey — COMMENTED** [review]
Any reason some files are renamed?

File: rke/templates/templates.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ const (
 	coreDnsv18  = "coredns-v1.8"
 	coreDnsv116 = "coredns-v1.16"
 	coreDnsv117 = "coredns-v1.17"
+	coreDnsv183 = "corends-v1.8.3"
```
Comment (superseb) [inline rke/templates/templates.go]: typo `corends`, als the numbers are corresponding to k8s versions but I guess thats not the same everywhere

Comment (kinarashah) [reply rke/templates/templates.go]: oops thanks! yeah it's not the same, this template would also work with 1.20.x + rbac rules changed for addon version so thought of going with 1.8.3

**Comment by kinarashah** [comment]
@StrongMonkey when I introduced them in earlier PR, the names were `319.go`, `046.go` which didn't give correct version info so renamed them to reflect that

**Review by StrongMonkey — APPROVED** [review]

**Review by superseb — APPROVED** [review]

**Pull request merged**